docs/data/material/components/material-icons/SearchIcons.js

```diff
@@ -149,21 +155,49 @@ function handleLabelClick(event) {
 
 function Icon(props) {
   const { icon, onOpenClick } = props;
+
+  const rootRef = React.useRef(null);
+  const [isVisible, setIsVisible] = React.useState(false);
+
+  React.useEffect(() => {
+    const margin = 200;
+    const observer = new IntersectionObserver(
+      (entries) => {
+        // `isIntersecting` is not working correctly on Chrome
+        const clientRect = entries[0].target.getBoundingClientRect();
+        const isIntersecting =
+          clientRect.top < window.innerHeight + margin &&
+          clientRect.bottom > -margin;
+        setIsVisible(isIntersecting);
+      },
+      { rootMargin: `${margin}px 0px` },
+    );
+    observer.observe(rootRef.current);
+    return () => {
+      observer.disconnect();
+    };
+  }, []);
+
   /* eslint-disable jsx-a11y/click-events-have-key-events */
   return (
     <StyledIcon
       key={icon.importName}
+      ref={rootRef}
       onClick={handleIconClick}
       data-icon-theme={icon.theme}
       data-icon-name={icon.name}
     >
-      <StyledSvgIcon
-        component={icon.Component}
-        tabIndex={-1}
-        onClick={onOpenClick}
-        title={icon.importName}
-      />
-      {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions -- TODO: a11y */}
+      {isVisible ? (
```

Comment:
I think we should skip this for the first icons, they will be in the viewport. This will avoid the page rendering in two steps.